### ISSUE TITLE
Build revised donation form UI

### ISSUE THREAD

**Issue by lovegushwa** [mozilla/donate.mozilla.org#436]
@ScottDowne @adamlofting @valianttry @edrushka 
Hi Scott, can you  please build out the new donation form UI, detailed UI for each page including markup coming shortly. I will create a separate ticket for the thank you page (2nd email opt in and social share)

![screen shot 2015-09-23 at 2 43 54 pm](https://cloud.githubusercontent.com/assets/12228435/10059734/913ae022-6201-11e5-9743-675ac9aaf8b0.png)

**Comment by lovegushwa** [comment]
@cassiemc @vazquez 
R? ... I folded in today's critique feedback and Germaniaifyd the text. In addition you will see a super subtle light color in the fields that help make them pop a bit (rather then being filled with white.)

Once these are good to go i'll do markup for @ScottDowne hopefully early tomorrow morning.

One thing to note, on slide 3, we don't have a translation for "We want to include you in our work"
@alicoding any thoughts on how to get this? We have similar messages that could work, but they are much to long. 



![screen shot 2015-09-23 at 4 54 31 pm](https://cloud.githubusercontent.com/assets/12228435/10062105/fa265c80-6213-11e5-9a1f-41ff1d10cfc9.png)


![screen shot 2015-09-23 at 4 54 44 pm](https://cloud.githubusercontent.com/assets/12228435/10062111/07999288-6214-11e5-83ab-b22c107e4757.png)


![screen shot 2015-09-23 at 4 55 04 pm](https://cloud.githubusercontent.com/assets/12228435/10062124/265b11d8-6214-11e5-9165-072c9c39fd66.png)

**Comment by lovegushwa** [comment]
I just noticed on my mac screen I might need to add a bit more color inside the fields.

**Comment by cassiemc** [comment]
R+. The only things I'd consider are possibly making the "Weiter" button about 30% wider on the first and second screens, because with the handwritten type it seems a bit lower in the hierarchy. But it's not a deal-breaker. Last consideration might be the make the icons in the fields more subtle, but again not a deal-breaker. The color of the fields imo is fine, nice and subtle.

**Comment by lovegushwa** [comment]
@cassiemc thank you! ... I did make the icons a bit lighter (they were black) but I will punch down a bit more. Good call on the "Weiter" button :)

**Comment by lovegushwa** [comment]
@ScottDowne here is the first markup page of the new funnel. I'll keep them coming tomorrow but this should give you enough to get building. You can download assets below: 

Das hilft uns wirklich weiter!
https://drive.google.com/a/mozilla.com/file/d/0B2qu3ExKWIlvTHc5YzBPX3Judjg/view?usp=sharing

Red arrow
https://drive.google.com/a/mozilla.com/file/d/0B2qu3ExKWIlva3lTd0Z6ZThyVFk/view?usp=sharing

Mozilla
https://drive.google.com/a/mozilla.com/file/d/0B2qu3ExKWIlvR0VEeEdnZGlCSHM/view?usp=sharing


Page 1 of 5
<img width="996" alt="screen shot 2015-09-30 at 11 52 12 pm" src="https://cloud.githubusercontent.com/assets/12228435/10214588/77258174-67ce-11e5-9365-3ef46793d478.png">
<img width="997" alt="screen shot 2015-09-30 at 11 52 26 pm" src="https://cloud.githubusercontent.com/assets/12228435/10214589/7c55b3bc-67ce-11e5-8df9-64341cef501b.png">
<img width="997" alt="screen shot 2015-09-30 at 11 52 48 pm" src="https://cloud.githubusercontent.com/assets/12228435/10214591/836191b2-67ce-11e5-9dbe-a2bd01e34209.png">

**Comment by lovegushwa** [comment]
<img width="997" alt="screen shot 2015-10-05 at 2 49 31 pm" src="https://cloud.githubusercontent.com/assets/12228435/10294669/9bd1f2f0-6b70-11e5-84fe-b61b9e6c1181.png">
<img width="993" alt="screen shot 2015-10-05 at 2 49 11 pm" src="https://cloud.githubusercontent.com/assets/12228435/10294671/a0801d54-6b70-11e5-87aa-1bafd9ed7611.png">
<img width="994" alt="screen shot 2015-10-05 at 2 49 45 pm" src="https://cloud.githubusercontent.com/assets/12228435/10294675/a5988696-6b70-11e5-9eac-56fafb907424.png">

**Comment by lovegushwa** [comment]
<img width="996" alt="screen shot 2015-10-05 at 2 53 08 pm" src="https://cloud.githubusercontent.com/assets/12228435/10294729/f9a72abc-6b70-11e5-93b5-613e731ab297.png">
<img width="995" alt="screen shot 2015-10-05 at 2 53 32 pm" src="https://cloud.githubusercontent.com/assets/12228435/10294737/fdc0ca36-6b70-11e5-8bde-3604bb385fbb.png">
<img width="995" alt="screen shot 2015-10-05 at 2 53 49 pm" src="https://cloud.githubusercontent.com/assets/12228435/10294742/00b4594c-6b71-11e5-9cb1-754f613bbf34.png">

**Comment by lovegushwa** [comment]
<img width="997" alt="screen shot 2015-10-05 at 2 55 28 pm" src="https://cloud.githubusercontent.com/assets/12228435/10294783/417e8db2-6b71-11e5-8535-5bf91268bb56.png">
<img width="998" alt="screen shot 2015-10-05 at 2 55 46 pm" src="https://cloud.githubusercontent.com/assets/12228435/10294790/4a7b18d6-6b71-11e5-8cb2-ab54bf05835b.png">
<img width="997" alt="screen shot 2015-10-05 at 2 56 05 pm" src="https://cloud.githubusercontent.com/assets/12228435/10294794/4ff142c2-6b71-11e5-94aa-9d61db0a3c5c.png">

**Comment by lovegushwa** [comment]
<img width="997" alt="screen shot 2015-10-05 at 2 57 51 pm" src="https://cloud.githubusercontent.com/assets/12228435/10294844/938a39d0-6b71-11e5-9fd7-7f06e9189beb.png">
<img width="997" alt="screen shot 2015-10-05 at 2 58 08 pm" src="https://cloud.githubusercontent.com/assets/12228435/10294846/970e8bc4-6b71-11e5-9664-86b6cc8b0a7c.png">
<img width="997" alt="screen shot 2015-10-05 at 2 58 25 pm" src="https://cloud.githubusercontent.com/assets/12228435/10294852/99ad9cf8-6b71-11e5-93dd-652c91542008.png">

**Comment by lovegushwa** [comment]
@ScottDowne you can find any image you need to build the form's here:
https://drive.google.com/a/mozilla.com/folderview?id=0B2qu3ExKWIlvTWY3OHFBMlQwUHM&usp=sharing

**Comment by ScottDowne** [comment]
I'm struggling on where to start with this, probably questions for @adamlofting.

We could split it up, and test some of it. Some things seem obvious or simple enough that a test wouldn't be all that useful? I'm also concerned with splitting it up because some changes and made to go with others, so splitting it might not give it a fair chance.

I could do this all in one patch, and we can test it all in one test. One test with all these changes, and the current form as baseline. However, that gives us a pretty weird transitional period with two fiarly different forms.

We can make all these changes and simply not test them, but this seems like the least likely choice. To me most of these changes feel pretty logical, but people can react to things in unexpected ways...

If I were to split it up, my first choice would be:

don't test:
- colour changes
- size changes
- position changes
- border and border radius changes

do test:
- this really helps up arrow
- a signup checkbox on the final step
- removal of any input on the third step
- gradients (moving from flat ui to not flat ui)
- most of the thank-you/share page

**Comment by ScottDowne** [comment]
Thinking about this a bit more, and I think I'm putting together a plan.

A log of these changes are css only, so we can do a thing where the two versions of the page are toggled on and off. Would actually be pretty cool.

The changes that are not css.
- this really helps up arrow
- a signup checkbox on the final step
- removal of any input on the third step

Would be done via alternate components.

This gives us a fairly easy way to test all, some, or none of these changes. We can be flexible, and it's not very disruptive to code itself.

**Comment by adamlofting** [comment]
Here's my view (but final decision will be for @valianttry )

Run one UI test:

* I suggest we test all of the UI changes in one go using our [existing workflow for running a significant test](https://github.com/mozilla/donate.mozilla.org/blob/master/docs/Metrics.md#how-to-setup-a-new-ab-test-for-a-significantly-different-page-or-form). 
* This would include all color, spacing, gradient, imagery changes in one go. 
* Most likely, we want to role all these changes out from a design/brand perspective, and testing them all at oncce is a good sense check that we're not doing harm to overall conversion rate, without taking using up lots of testing capacity.
* *Eventually* we could test elements of these changes individually (like gradients vs flat) but these are minor tests in relation to fundraising asks so I wouldn't prioritise testing everyone one up front.

Then, we can individually test the the UX/flow changes:

* this really helps up arrow
* a signup checkbox on the final step
* removal of any input on the third step

**Comment by valianttry** [comment]
I am +1 for learning as much as we can about what converts in the shortest amount of time -- that way we can turn on the winning combination as soon as possible. 

@adamlofting or @ScottDowne  are there downsides I'm missing? If not.... LET'S ROLL.

**Comment by ScottDowne** [comment]
Landed the bulk of the changes:

https://donate.mofostaging.net/en-US/?test=new-flow-test
Test credit card number is: 4242424242424242 01/16 123

Old page still exists here: https://donate.mofostaging.net/en-US/

Things that have not happened yet:
- [ ] updated currency dropdown style.
- [ ] switch single and monthly, and add arrow. (can already preselect monthly)
- [ ] updated style on https://donate.mofostaging.net/en-US/one-page/ to be testable too. (It is possible we're already using one-page as our main by the time we test this, so we should have it ready)
- [ ] email opt in checkbox on the third page.
- [ ] arrow to the opt in checkbox.
- [ ] remove the bulk of address info. I read that zip code is much better that country. cvc/zip should be all we need for fraud reasons, I believe.

**Comment by edrushka** [comment]
Really exciting to see this starting to come together! What a great way to wrap up the week.

**Comment by ScottDowne** [comment]
@lovegushwa not sure what the new copy is in English, which I need to do this.

For now I'll use the old page's copy, and see how close I can make it match.

**Comment by ScottDowne** [comment]
Not a huge issue, but the "Wir bei Mozilla sagen vielen Dank, {name}!" bit doesn't have the bolded syntax in the source string, so the translation doesn't retain where to keep things bolded.

It also won't know where to newline it.

The English currently looks like "From all of us at Mozilla, thank you {name}!"

In comparison, Chinese formats it like this "我们 Mozilla 的所有人都非常感谢您，{name}！"

And Italian looks like this: "{name}, da parte di tutti noi di Mozilla: grazie mille!"

As you can see the "thank you {name}!" has moved around. For things like bold and explicit newlines, we need to define that as markup in the string itself, and the translators handle the placement of the style.

**Comment by lovegushwa** [comment]
@ScottDowne @edrushka Redpen QA comments here:
https://redpen.io/p/wae7651ddf284d3216

**Comment by ScottDowne** [comment]
Tests for share and thank-you are done:
https://donate.mofostaging.net/de/thank-you/?test=new-flow-test
https://donate.mofostaging.net/de/share/?test=new-flow-test

**Comment by ScottDowne** [comment]
We can configure the address now: https://donate.mofostaging.net/en-US/?test=address-test%20new-flow-test

Right now it's just country and postal/zip code.

**Comment by valianttry** [comment]
@ScottDowne I did a test, everything looks really great. Two quick things:
1. There's no default amount (maybe this is solved depending upon context; e.g. if someone selects an amount on snippet it is carried onto the form)
and
2. I noticed I have to enter my email on the donation form AND re-enter it on the email signup page. Can we auto-fill email for the optin page?

**Comment by ScottDowne** [comment]
@valianttry glad it works :)

1. amount is handled from a value in the url, which the snippet can adjust. It's not quite the same as a default value, but gets the same result for the moment. If we add a default amount, it would be for cases that didn't come from sources we control, like a google search, direct url typed, or a independently shared url. It is something I'm sure we'll get to at some point, but for now strikes me as lower priority.

2. email not being passed to thank-you is flagged in a checkbox list a few comments up. It needs to happen before this ticket is considered resolved, which should be before the 15th.

**Comment by lovegushwa** [comment]
@ScottDowne
Please get the snippet and funnel "this really helps us" monthly donation ask, and email signup funnel graphics here (will localize email ask for page 3 of the funnel after EN test runs):
https://drive.google.com/a/mozilla.com/folderview?id=0B2qu3ExKWIlvakhPY05aamRqUjQ&usp=sharing

![screen shot 2015-11-23 at 10 58 54 am](https://cloud.githubusercontent.com/assets/12228435/11346409/60b33da0-91d1-11e5-81d6-832acb468288.png)

**Comment by adamlofting** [comment]
My bad... I'd forgotten to file a testing ticket for this. I'd not forgotten the new UI though :+1:

**Comment by ScottDowne** [comment]
@lovegushwa Am I missing something with the above images? Are they in google drive?

Sorry if I missed them, but I have a bunch of rando things on my plate.

**Comment by ScottDowne** [comment]
Lol, I see the drive link now.

**Comment by adamlofting** [comment]
This is a long ticket, and I'm not sure if there are any remaining dev tasks, but we're ready to make this New UI the default for all traffic now. I've file that in #1049 so we might be able to close this.
